server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -375,32 +284,26 @@ public SourceFieldMapper build() {
 
     }
 
-    private static SourceFieldMapper resolveSourceMode(final IndexMode indexMode, final Mode sourceMode, boolean enableRecoverySource) {
+    private static SourceFieldMapper resolveSourceMode(final IndexMode indexMode, final Mode sourceMode) {
         switch (indexMode) {
             case STANDARD:
                 switch (sourceMode) {
                     case SYNTHETIC:
-                        return enableRecoverySource ? DEFAULT_SYNTHETIC : DEFAULT_SYNTHETIC_NO_RECOVERY_SOURCE;
+                        return DEFAULT_SYNTHETIC;
                     case STORED:
-                        return enableRecoverySource ? DEFAULT : DEFAULT_NO_RECOVERY_SOURCE;
+                        return DEFAULT;
                     case DISABLED:
-                        return enableRecoverySource ? DEFAULT_DISABLED : DEFAULT_DISABLED_NO_RECOVERY_SOURCE;
+                        return DEFAULT_DISABLED;
                     default:
                         throw new IllegalArgumentException("Unsupported source mode: " + sourceMode);
                 }
             case TIME_SERIES:
```

Comment:
Move the values from below to here, it's weird that we keep checking for TIME_SERIES after the `case`.

Reply:
whoops, I missed this. I will take care of this the follow that removes the singleton instances for the different index modes.